src/sentry/analytics/event.py

```diff
@@ -14,6 +16,32 @@ def __init__(self, name, type=six.text_type, required=True):
         self.required = required
 
 
+class Map(Attribute):
+    def __init__(self, name, attributes, required=True):
+        self.name = name
+        self.required = required
+        self.attributes = attributes
+
+    def type(self, value):
+        if not isinstance(value, dict):
+            raise ValueError('Value must be a dictionary')
+
+        data = {}
+        for attr in self.attributes:
+            if attr.required:
```

Comment:
This assumes `None` is a valid value for a required field which seems questionable, depending on the context